content/reference-transaction-format.md

```diff
@@ -25,6 +26,63 @@ Any signature type can authorize any type of transaction, with the following exc
+
+### Master Keys
+
+When you create an account using the [`wallet_propose`](reference-rippled.html#wallet-propose) method, the `master_seed` (private key) and `public_key` returned in the response are the **master keys** for the account. These master keys are an intrinsic part of the account's identity as the `account_id` (account address), also returned in the `wallet_propose` response, and the `public_key` value are mathematically related.
+
+Here's a sample `wallet_propose` response, including the `master_seed` (private key), `public_key`, and `account_id` (account address) values:
+
+```
+{
+  "result": {
+    "account_id": "rM8oJwkKtgEU2HZkCHsJScLHqegWpgBRxt",
+    "key_type": "secp256k1",
+    "master_key": "SAY ARMY KITE LETS PAN WAVE JUKE ACRE MAT TILT FEAT LED",
+    "master_seed": "PrivateKeyXXXXXXXXXXXXXXXXXXX",
+    "master_seed_hex": "47E8F7F59D0CC9A3F01FD3D69CCAE93A",
+    "public_key": "aBQEK3r6hb8isnCWmortkNdv3MSMKP74o7gj3kHZQF9vDkbXo4Qs",
+    "public_key_hex": "032D911AE9887278110A8FE25169546D4E95A82AF5EFA25525FC3FDFFDC5A8B604"
+  },
+  "status": "success",
+  "type": "response"
+}
+```
+
+Master keys have more powerful signing authority than regular keys. In addition to being able to sign all transactions that regular keys can, master keys are the only keys that can be used to perform the following actions:
+
+* Use the master private key to [disable the master public key](#accountset-flags).
+* Permanently give up the ability to [freeze counterparties](concept-freeze.html#no-freeze).
+
+**Important:** Master keys can be disabled, but cannot be changed. This means that if your master private key is [compromised](concept-issuing-and-operational-addresses.html), you cannot change it to stop the compromise.
+
+This is a compelling reason to keep your master keys offline whenever possible and set up regular keys to sign transactions from your master account instead.
+
+
+### Regular Keys
+
+When you create an account using the [`wallet_propose`](reference-rippled.html#wallet-propose) method, the `master_seed` (private key) and `public_key` returned in the response are the master keys for the account. For the purposes of this discussion, let's call this account the master account. Ripple recommends that you keep the master account's master keys offline whenever possible and assign regular keys to the master account to sign your transactions instead.
+
+You generate regular keys using the same `wallet_propose` method used to generate a master account and master keys. However, you use this second account and key pair specifically with the `SetRegularKey` method to assign **regular keys** to the master account.
+
+Once you've assigned the regular keys, the master account has two key pairs associated with it:
+
+* Master keys that are intrinsically related to its address and which you keep offline whenever possible.
```

Comment:
It's not totally clear what you mean by "keep offline whenever possible" here. To be clear, what this means is "don't put your master private key somewhere malicious actors can get access to it".

That can mean an airgapped machine that never, ever connects to the internet. That can mean a piece of paper printed out and stored in a safe. That can mean, "on your personal computer behind a firewall". Generally it means, "Not within reach of a computer program that interacts with the internet at large." And definitely not as a tattoo on your forehead.

Reply:
Also, for clarity, it's safe to share your public key with anyone and everyone, even your master public key. In fact, the moment your account sends its first transaction, your master public key _will_ be shared with everyone on the entire XRP Ledger, because that's part of how you prove you're the one who owns that account.

However, _before_ you send your first transaction, you can optionally keep your public key secret as an extra security measure. This means you can't have any regular keys or anything else set (gotta send a transaction to do that) but instead of having one seemingly-unsolvable math problem protecting your account, you have two. For an attacker to send a transaction as you when you haven't revealed your public key, first they'd have to go from a hash of your public key to your public key; then they'd have to go from your public key to a private key. This is less crazy than it sounds because it's possible that advances in cryptography/math make the core problems in _either_ the signing algorithm (ECDSA) or the hash functions (RIPEMD160 of SHA256) less "seemingly impossible" to solve.